c/helper.c

```diff
@@ -13,10 +13,10 @@ int buffer_append(void *ext, unsigned char *str, size_t size)
 {
 	buffer_t *b = ext;
 	int new_size, new_used;
-	char *tmp;
+	unsigned char *tmp;
 
 	new_used = b->used + size;
-	for (new_size = b->size || 8; new_size < new_used; new_size *= 2);
+	for (new_size = 1; new_size < new_used; new_size *= 2);
```

Comment:
It's not obvious why this is a "fix warnings"-level change, this looks like it actually changes behavior unless we somehow know that `b->size` is always 1.